src/Compilers/Core/Portable/MetadataReader/PEAssembly.cs

```diff
@@ -75,6 +75,8 @@ internal PEAssembly(AssemblyMetadata owner, ImmutableArray<PEModule> modules)
             _owner = owner;
         }
 
+        internal AssemblyMetadata Metadata => _owner;
```

Comment:
>internal AssemblyMetadata Metadata => _owner; [](start = 8, length = 45)

Do we really have to expose original object? It looks like we were not comfortable doing this in the past. Can we rely on ```GetNonDisposableMetadata``` API instead?